spec/observables/forkJoin-spec.ts

```diff
@@ -595,5 +611,164 @@ describe('forkJoin', () => {
         },
       });
     });
+
+    it('concat reference', () => {
+      const results: Array<number | string> = [];
+      concat(of(1), of(2))
+        .pipe(last())
+        .subscribe({
+          next(value) {
+            results.push(+value);
+          },
+          error(err) {
+            throw err;
+          },
+          complete() {
+            results.push('done');
+          },
+        });
+      expect(results).to.deep.equal([2, 'done']);
+    });
+
+    it('concatMap reference', () => {
+      const results: Array<number | string> = [];
+      of(1)
+        .pipe(concatMap((val) => of(2)))
+        .subscribe({
+          next(value) {
+            results.push(+value);
+          },
+          error(err) {
+            throw err;
+          },
+          complete() {
+            results.push('done');
+          },
+        });
+      expect(results).to.deep.equal([2, 'done']);
+    });
+
+    xit('should work with concatMap fail', () => {
+      const results: Array<number | string> = [];
+      forkJoin([of(1).pipe(concatMap((val) => of(2))), of(3).pipe(concatMap((val) => of(4)))]).subscribe({
+        next(value) {
+          results.push(+value);
+        },
+        error(err) {
+          throw err;
+        },
+        complete() {
+          results.push('done');
+        },
+      });
+      expect(results).to.deep.equal([2, 4, 'done']);
+    });
+
+    xit('should work with concatMap cross types fail', () => {
+      const results: Array<number | string> = [];
+      forkJoin([
+        of('a').pipe(concatMap((val: string) => of(1))) as Observable<number>,
+        of('b').pipe(concatMap((val: string) => of(2))) as Observable<number>,
+      ]).subscribe({
+        next(value) {
+          results.push(+value);
+        },
+        error(err) {
+          throw err;
+        },
+        complete() {
+          results.push('done');
+        },
+      });
+      expect(results).to.deep.equal([1, 2, 'done']);
+    });
+
+    it('should work with concatMap cross types mergeAll', () => {
+      const results: Array<number | string> = [];
+      forkJoin([
+        of('a').pipe(concatMap((val: string) => of(1))) as Observable<number>,
+        of('b').pipe(concatMap((val: string) => of(2))) as Observable<number>,
+      ])
+        .pipe(mergeAll())
+        .subscribe({
+          next(value) {
+            results.push(+value);
```

Comment:
Here we're pushing `+[1, 2]` onto the array. Basically converting the array to `NaN`. So this is incorrect, and we'd have to push both values into the array, or stop converting it to a number, and push the whole array in, then change the expectation to be `[[1, 2], 'done']` below.